core/src/main/java/dev/keva/core/command/impl/zset/ZAdd.java

```diff
@@ -0,0 +1,123 @@
+package dev.keva.core.command.impl.zset;
+
+import dev.keva.core.command.annotation.CommandImpl;
+import dev.keva.core.command.annotation.Execute;
+import dev.keva.core.command.annotation.Mutate;
+import dev.keva.core.command.annotation.ParamLength;
+import dev.keva.ioc.annotation.Autowired;
+import dev.keva.ioc.annotation.Component;
+import dev.keva.protocol.resp.reply.BulkReply;
+import dev.keva.protocol.resp.reply.ErrorReply;
+import dev.keva.protocol.resp.reply.IntegerReply;
+import dev.keva.protocol.resp.reply.Reply;
+import dev.keva.store.KevaDatabase;
+import dev.keva.util.DoubleUtil;
+import dev.keva.util.hashbytes.BytesKey;
+
+import java.nio.charset.StandardCharsets;
+import java.util.AbstractMap.SimpleEntry;
+
+import static dev.keva.util.Constants.FLAG_CH;
+import static dev.keva.util.Constants.FLAG_GT;
+import static dev.keva.util.Constants.FLAG_INCR;
+import static dev.keva.util.Constants.FLAG_LT;
+import static dev.keva.util.Constants.FLAG_NX;
+import static dev.keva.util.Constants.FLAG_XX;
+
+@Component
+@CommandImpl("zadd")
+@ParamLength(type = ParamLength.Type.AT_LEAST, value = 3)
+@Mutate
+public final class ZAdd {
```

Comment:
I think the new commands should have a separate branch and PR for them
missing RW lock change with new command feature might be too much to review in 1 PR

Reply:
oh nvm, it's already merge in another PR, this is just branching mixed